### PR TITLE
Add a colection of ESLint plugins

### PR DESCRIPTION
**Pull request by RabsRincon**
This adds various ESLint plugins such as:
 - [eslint-plugin-jsdoc](https://github.com/gajus/eslint-plugin-jsdoc/issues)
 - [eslint-plugin-mocha](https://github.com/lo1tuma/eslint-plugin-mocha)
 - [eslint-plugin-node](https://github.com/mysticatea/eslint-plugin-node)
 - [eslint-plugin-requirejs](https://github.com/cvisco/eslint-plugin-requirejs)

It also bumps the required node version on `package.json` to >= 12 as specified on the documentation.
It also introduces ESLint rules to the `test/` folder - Mostly the same as in the serverside one.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -103,11 +110,13 @@
...
-    "ci-lint": "eslint --config static/.eslintrc --ignore-path static/.eslintignore ./static/*.js ./static/panes/*.js ./static/modes/*.js && eslint --config .eslintrc --ignore-path .eslintignore ./app.js ./lib/**",
+    "ci-lint": "npm run lint",
     "ci-test": "nyc --reporter=lcov mocha --recursive",
-    "lint-client": "eslint --config static/.eslintrc --ignore-path static/.eslintignore --fix ./static/*.js ./static/panes/*.js ./static/modes/*.js",
-    "lint-server": "eslint --config .eslintrc --ignore-path .eslintignore --fix ./app.js ./lib/**",
-    "lint": "npm run lint-client && npm run lint-server",
+    "lint-client": "eslint --config static/.eslintrc --ignore-path static/.eslintignore ./static/*.js ./static/panes/*.js ./static/modes/*.js",
+    "lint-server": "eslint --config .eslintrc --ignore-path .eslintignore ./app.js ./lib/**",
+    "lint-tests": "eslint --config test/.eslintrc --ignore-path test/.eslintignore ./test/**/*.js ./test/*.js",
+    "lint": "npm run lint-client && npm run lint-server && npm run lint-tests",
```
Comment (apmorton) [inline package.json:118]: the help string in `make lint` implies `--fix`, which is no longer the case.

We should either update the help string in the makefile or restore the `--fix` behavior of `make lint`

**Comment by RabsRincon** [comment]
I also wanted to add the `quotes` rule to enforce a consistent style - But didn't know which one we prefer. It seems to be evenly split between double quoutes and simple ones. Any suggestions?

**Comment by apmorton** [comment]
#singlequotemasterrace

**Comment by partouf** [comment]
> I also wanted to add the `quotes` rule to enforce a consistent style - But didn't know which one we prefer. It seems to be evenly split between double quoutes and simple ones. Any suggestions?

I really have no idea. Single quotes usually look nicer, but still, I'm on the C++ side of things where strings are double quotes and single characters are single quotes. But in practice, I actually usually just look what's used around the code that I'm changing and conform it to that... So I don't know.

**Comment by apmorton** [comment]
@RabsRincon lgtm besides the `make lint` bit I mentioned above

**Pull request closed without merging**